libs/dh/metering-point/feature-upload-measurements/src/upload-page.ts

```diff
@@ -235,11 +221,23 @@ export class DhMeasurementsUploadComponent {
     }
   });
 
-  csv = signal<CsvParseResult | null>(null);
+  csv = signal<MeasureDataResult | null>(null, { equal: () => false });
+  totalSum = computed(() => this.csv()?.sum ?? 0);
+  totalPositions = computed(() => this.csv()?.measurements.length ?? 0);
+  progress = computed(() => (this.csv()?.isFatal ? undefined : this.csv()?.progress));
+  quality = computed(() => {
+    const qualities = this.csv()?.qualities;
+    if (!qualities?.size) return null;
+    const [first] = qualities;
+    return qualities.size > 1 ? 'MIXED' : first;
+  });
+
   private validate = async (): Promise<ValidationErrors | null> => {
     if (!this.file.value) return null;
     const [file] = this.file.value;
-    await this.csvParser.parseFile(file).forEach(this.csv.set);
+    const resolution = this.resolution();
+    assertIsDefined(resolution);
+    await this.measurements.parseFile(file, resolution).forEach(this.csv.set);
```

Comment:
praise: I was about to write a comment how `forEach` won't work here, but then TIL it works! because `forEach` here is from RxJS, and not the native JS method. Nicely done 💯